packages/components/managed-data-table/src/data-table-settings/data-table-settings.js

```diff
@@ -0,0 +1,189 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import requiredIf from 'react-required-if';
+import { useIntl } from 'react-intl';
+import Constraints from '@commercetools-uikit/constraints';
+import SelectInput from '@commercetools-uikit/select-input';
+import Spacings from '@commercetools-uikit/spacings';
+import DensityManager from './density-manager';
+import ColumnManager from './column-manager';
+import TABLE_SETTINGS_ACTIONS, {
+  COLUMN_MANAGER,
+  DISPLAY_SETTINGS,
+} from '../constants';
+import messages from './messages';
+
+export const getDropdownOptions = ({
+  isColumnManagerEnabled,
+  isDisplaySettingsEnabled,
+  formatMessage,
+}) => [
+  ...(isColumnManagerEnabled
+    ? [
+        {
+          value: COLUMN_MANAGER,
+          label: formatMessage(messages.columnManagerOption),
+        },
+      ]
+    : []),
+  ...(isDisplaySettingsEnabled
+    ? [
+        {
+          value: DISPLAY_SETTINGS,
+          label: formatMessage(messages.displaySettingsOption),
+        },
+      ]
+    : []),
+];
+
+export const getMappedColumns = (columns = []) =>
+  columns.reduce(
+    (mappedColumns, column) => ({
+      ...mappedColumns,
+      [column.key]: column,
+    }),
+    {}
+  );
+
+export const getSelectedColumns = (visibleColumnsKeys = [], mappedColumns) =>
+  visibleColumnsKeys.map((columnKey) => mappedColumns[columnKey]);
+
+const DataTableSettings = (props) => {
+  const intl = useIntl();
+  const [openPanel, setOpenPanel] = useState(null);
```

Comment:
The name is a bit confusing. I thought at first this was a boolean flag. Instead it's a string value. Can we have a more specific name to avoid confusion?
Thanks

Reply:
Do you think `openedPanelId` would be better fit?

Reply:
Yes better thanks